bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/ConfigOptionProvider.java

```diff
@@ -29,19 +29,11 @@
 @NonNullByDefault
 public interface ConfigOptionProvider {
 
-    /**
-     * Provides a collection of {@link ParameterOptions}s.
-     *
-     * @deprecated Use {@link getParameterOptions} with context instead.
-     *
-     * @param uri the uri of the config description
-     * @param param the parameter name for which the requested options shall be returned
-     * @param locale the locale in which the result is expected
-     * @return the configuration options provided by this provider if any or {@code null} otherwise
-     */
-    @Deprecated
-    @Nullable
-    Collection<ParameterOption> getParameterOptions(URI uri, String param, @Nullable Locale locale);
+    // @Deprecated
+    // @Nullable
+    // default Collection<ParameterOption> getParameterOptions(URI uri, String param, @Nullable Locale locale) {
```

Comment:
This method needs a little attention. If we remove it now it will break the openHAB 3 Add-ons build.

Reply:
So if I get it right, your plan now is to keep it as a deprecated "default" method in place and remove it later?
Which add-ons will break?

Reply:
Yes, that was the initial idea. We are talking about three bindings:

- Amazon Dash Button
- GPSTracker
- SmartMeter

There are two other options:

1. We could exclude them from openHAB 3 Add-ons build.

or

2. We could move them to "master"-branch and migrate them right now.

Reply:
> We are talking about three bindings

Shouldn't it be fairly easy to adapt them?

Reply:
> Shouldn't it be fairly easy to adapt them?

Yes, it is. But we cannot use the same code to compile against OHC 2.5 and OHC 3 (after this PR got merged). If we remove the deprecated method in the bindings it will fail on OH 2.5.x Add-ons build and if we remove the deprecated method here it will fail on OH 3 Add-ons build.

Reply:
Ok, I see. I had missed the fact that it is a "provider" interface and not a "consumer" interface.
So yeah, I think your approach is just fine - let's leave it in as a default method and remove it once we move all add-ons to master. Do you keep track of those open changes somewhere?

Reply:
We keep track of it in https://github.com/openhab/openhab-core/issues/1408. I added a note.